src/mem/external_master.hh

```diff
@@ -115,19 +124,36 @@ class ExternalMaster : public SimObject
     static std::map<std::string, Handler *> portHandlers;
 
   public:
+    /**
+      * @ingroup api_port
+      */
     ExternalMaster(ExternalMasterParams *params);
 
-    /** Port interface.  Responds only to port "port" */
+    /**
+     * Port interface.  Responds only to port "port"
+     *
+     * @ingroup api_port
+     */
     Port &getPort(const std::string &if_name,
                   PortID idx=InvalidPortID) override;
 
-    /** Register a handler which can provide ports with port_type ==
-     *  handler_name */
+    /**
+     * Register a handler which can provide ports with port_type ==
+     * handler_name
+     *
+     * @ingroup api_port
+     */
     static void registerHandler(const std::string &handler_name,
         Handler *handler);
 
+    /**
+      * @ingroup api_port
+      */
     void init() override;
 
+    /**
+      * @ingroup api_port
+      */
     const MasterID masterId;
```

Comment:
Should this be public? Is it used publicly anywhere?

Reply:
Removed masterId from the API.

Reply:
anyway, a function in util/tlm/src/sc_master_port.cc  uses masterId to make a request